### PR TITLE
Tasks

### PR DESCRIPTION
**Pull request by DimaMakarenko**

### CODE REVIEW

File: app/screens/Tasks/List/TaskRow.tsx [hunk 1]
```diff
@@ -1,74 +1,91 @@
...
+    navigate('Edit', { task, deleteTask, handleEdit, handlePause, handleStart });
+  }, [task, navigate, deleteTask, handlePause, handleStart]);
+
+  const handleShow = useCallback(() => {
+    navigate('Show', { taskId: task.id, deleteTask, handleEdit, handlePause, handleStart });
+  }, [task, deleteTask, handleEdit, handlePause, handleStart, navigate]);
 
-  const showAlert = () => alert('Deleting task', 'You really want delete this task?', handleDelete);
+  const showAlert = useCallback(() => {
+    isActive ? stopActiveTask() : alert('Deleting task', 'Are you really want delete this task?', handleDelete);
```
Comment (CheeBam) [inline app/screens/Tasks/List/TaskRow.tsx:49]: как-то странно)

File: app/screens/Tasks/List/TaskRow.tsx [hunk 2]
```diff
@@ -22,17 +22,25 @@ const TaskRow: React.FC<ITaskRow> = ({ task, navigate }) => {
...
+  }, [task, pauseTask]);
 
   const handleStart = useCallback(() => {
     startTask(task);
-  }, [task]);
+  }, [task, startTask]);
 
-  const showAlert = () => alert('Deleting task', 'You really want delete this task?', handleDelete);
+  const showAlert = useCallback(() => {
+    alert('Deleting task', 'You really want delete this task?', handleDelete);
```
Comment (art1990) [inline app/screens/Tasks/List/TaskRow.tsx]: repeating logic

File: app/screens/Tasks/Show/ShowTask.tsx [hunk 3]
```diff
@@ -18,18 +18,20 @@ interface IShowTask {
...
-  };
+  }, [navigation]);
 
   const lastEnd = useMemo(() => {
     return lastSessionEnd(timeSession, isActive);
   }, [timeSession, isActive]);
 
-  const showAlert = () => alert('Deleting task', 'You really want delete this task?', handleClick);
+  const showAlert = useCallback(() => {
+    alert('Deleting task', 'You really want delete this task?', handleDelete);
```
Comment (art1990) [inline app/screens/Tasks/Show/ShowTask.tsx:49]: repeating logic

File: app/screens/Tasks/List/ListTask.tsx [hunk 4]
```diff
@@ -57,7 +58,9 @@ const ListTask: FC<IListTask> = ({ navigation }) => {
                 <FlatList
                   data={tasks}
                   keyExtractor={(item) => item.id.toString()}
-                  renderItem={({ item }: { item: any }) => <TaskRow task={item} navigate={navigation.navigate} />}
+                  renderItem={({ item }: { item: any }) => (
```
Comment (art1990) [inline app/screens/Tasks/List/ListTask.tsx:61]: item : ITask ??

File: app/hooks/useTaskAction.ts [hunk 5]
```diff
@@ -70,12 +70,20 @@ export const useTaskAction = () => {
     [dispatch, uid],
   );
 
+  const deleteTask = useCallback(
+    async (taskId: number) => {
```
Comment (art1990) [inline app/hooks/useTaskAction.ts:76]: taskID number???

File: app/screens/Tasks/Show/ShowTask.tsx [hunk 6]
```diff
@@ -37,9 +44,32 @@ const ShowTask: React.FC<IShowTask> = ({ navigation, route }) => {
...
+          <View style={styles.icons}>
+            <TouchableOpacity onPress={() => handleEdit()} style={styles.optionIcon}>
+              <SvgUri source={editImg} />
+            </TouchableOpacity>
+            {isActive ? (
+              <TouchableOpacity style={styles.optionIcon} onPress={() => handlePause()}>
+                <SvgUri source={pauseImg} />
+              </TouchableOpacity>
+            ) : (
+              <TouchableOpacity style={styles.optionIcon} onPress={() => handleStart()}>
```
Comment (art1990) [inline app/screens/Tasks/Show/ShowTask.tsx:70]: your dont need callback in to onPress

File: app/hooks/useTaskAction.ts [hunk 7]
```diff
@@ -79,23 +79,41 @@ export const useTaskAction = () => {
...
+        },
+      };
+      await updateTaskDb(updates);
+    },
+    [uid],
+  );
 
   const getTask = useCallback(
     (id: number) => {
       return taskList.filter((task) => task.id === id)[0];
```
Comment (art1990) [inline app/hooks/useTaskAction.ts:98]: i think you can use find instead filter

**Pull request merged**